LogMonitor/src/LogMonitor/SystemInfo.cpp

```diff
@@ -0,0 +1,172 @@
+//
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+#include "pch.h"
+
+SystemInfo::SystemInfo() {
+    HKEY hKey;
+    std::wstring strValueOfBinDir;
+
+
+    LONG lRes = RegOpenKeyExW(
+        HKEY_LOCAL_MACHINE, REG_KEY_CUR_VER_STR, 0, KEY_READ, &hKey);
+    switch (lRes) {
+        case ERROR_SUCCESS:
+         Utility::GetStringRegKey(
+            hKey,
+            BUILD_BRANCH_STR_VALUE_NAME,
+            mRegistryCurrentVersion.BuildBranch,
+            REG_KEY_STR_DEFAULT_VALUE);
+         Utility::GetStringRegKey(
+            hKey,
+            BUILD_LAB_STR_VALUE_NAME,
+            mRegistryCurrentVersion.BuildLab,
+            REG_KEY_STR_DEFAULT_VALUE);
+         Utility::GetStringRegKey(
+            hKey,
+            CURRENT_BUILD_NUMBER_STR_VALUE_NAME,
+            mRegistryCurrentVersion.CurrentBuildNumber,
+            REG_KEY_STR_DEFAULT_VALUE);
+        Utility::GetStringRegKey(
+            hKey,
+            INSTALLATION_TYPE_STR_VALUE_NAME,
+            mRegistryCurrentVersion.InstallationType,
+            REG_KEY_STR_DEFAULT_VALUE);
+        Utility::GetDWORDRegKey(
+            hKey,
+            CURR_MINOR_VER_NUM_STR_VALUE_NAME,
+            mRegistryCurrentVersion.CurrentMinorVersionNumber,
+            REG_KEY_DW_DEFAULT_VALUE);
+        Utility::GetDWORDRegKey(
+            hKey,
+            CUR_MAJOR_VER_NUM_STR_VALUE_NAME,
+            mRegistryCurrentVersion.CurrentMajorVersionNumber,
+            REG_KEY_DW_DEFAULT_VALUE);
+        Utility::GetStringRegKey(
+            hKey,
+            PRODUCT_NAME_STR_VALUE_NAME,
+            mRegistryCurrentVersion.ProductName,
+            REG_KEY_STR_DEFAULT_VALUE);
+        break;
+    case ERROR_FILE_NOT_FOUND:
+        logWriter.TraceError(
+            Utility::FormatString(L"Key not found").c_str());
+        break;
+    default:
+        logWriter.TraceError(
+            Utility::FormatString(L"Error opening key").c_str());
+        break;
+    }
+
+    TCHAR buffer[256] = TEXT("");
+    DWORD dwSize = _countof(buffer);
+    int cnf = 0;
+    for (cnf = 0; cnf < ComputerNameMax; cnf++) {
+        if (!GetComputerNameEx((COMPUTER_NAME_FORMAT)cnf, buffer, &dwSize)) {
```

Comment:
I think Computer Name makes the telemetry pseudo-anonymous, we'll be treading PII grounds.
BTW, could you present a short proposal listing all the telemetry items we want to collect and the reason. I was of the opinion that we go for as a little as possible, just to get us to know the tool usage.

Reply:
We have this document out with Lucille as we initiate [privacy requirements](https://microsofteur.sharepoint.com/:w:/r/teams/DPLATWindowsContainersNRB-WindowsContainersOfficialChannel/_layouts/15/Doc.aspx?sourcedoc=%7B36CFB84E-F32F-413E-850E-3FF2B8AFEADD%7D&file=LogMonitor%20Privacy%20Review.docx&action=default&mobileredirect=true&DefaultItemOpen=1&share=IQFOuM82L_M-QYUOP_K4r-rdAWSsvx2YLr-lThGYYpcfVFM) discussions. I am trying to to answer the why, how and reason for the data points we are targeting.

Reply:
Feel free to give your thoughts on that document. Also we are thinking to have as many data points as possible submitted for review so that next time we do not have to go through this process again.